internal/exec/engine.go

```diff
@@ -139,7 +139,12 @@ func (e Engine) Run(stageName string) error {
 	defer e.Logger.PopPrefix()
 
 	fullConfig := latest.Merge(baseConfig, latest.Merge(systemBaseConfig, cfg))
-	if err = stages.Get(stageName).Create(e.Logger, e.Root, *e.Fetcher).Run(fullConfig); err != nil {
+	err = stages.Get(stageName).Create(e.Logger, e.Root, *e.Fetcher).Run(fullConfig)
+	if err == resource.ErrNeedNet && stageName == "fetch-offline" {
+		err = SignalNeedNet()
+		// fall through
```

Comment:
Let's still do a `e.Logger.Crit()` here to be consistent with the other `SignalNeedNet()` error-handling above?